### ISSUE TITLE
[Feature Request] updateOrCreate

### ISSUE THREAD

**Issue by ksaitor** [balderdashy/waterline#790]
I guess it'll look like this:
```javascript
  updateOrCreate: function (criteria, data, cb) {
    var that = this;
    var resolver = Promise.pending();
    cb = cb || function () {};

    that.findOne(criteria).then(function (model) {
      if (model) {
        // Model exists. Update.
        model = _.deepExtend(model, data);
        model.save(function () {
          resolver.resolve(model);
          cb(null, model);
        });
      } else {
        // Model does not exist. Create.
        that.create(data).done(function () {
          resolver.resolve(model);
          cb(null, model);
        });
      }
    })
    .catch(function (err) {
      resolver.reject(err);
      cb(err);
    });

    return resolver.promise;
  }
```

**Comment by dmarcelino** [comment]
Hi @ksaitor, would you mind submitting a PR? Some useful links below:
* findOrCreate code: https://github.com/balderdashy/waterline/blob/master/lib/waterline/query/composite.js#L24
* findOrCreate test: https://github.com/balderdashy/waterline/blob/master/test/unit/query/query.findOrCreate.js

**Comment by dmarcelino** [comment]
PR #940 adds this, relevant comment: https://github.com/balderdashy/waterline/pull/940#issuecomment-93937764.

**Comment by sailsbot** [comment]
Thanks for posting, @ksaitor.  I'm a repo bot-- nice to meet you!

It has been 60 days since there have been any updates or new comments on this page.  If this issue has been resolved, feel free to disregard the rest of this message.  On the other hand, if you are still waiting on a patch, please:

  + review our [contribution guide](https://github.com/balderdashy/sails/blob/master/CONTRIBUTING.md) to make sure this submission meets our criteria (only _verified bugs_ with documented features, please;  no questions, commentary, or bug reports about undocumented features or unofficial plugins)
  + create a new issue with the latest information, including updated version details with error messages, failing tests, and a link back to [the original issue]().  This allows GitHub to automatically create a back-reference for future visitors arriving from search engines.

Thanks so much for your help!

**Comment by Tronix117** [comment]
This should be reopened, mysql (INSERT ON DUPLICATE) and postgresql (UPSERT) with fallbacks should be implemented

**Comment by AlecTaylor** [comment]
Anything happening with this?

BTW: dmarcelino link is broken, new link -https://github.com/balderdashy/waterline/blob/master/lib/waterline/methods/find-or-create.js
